src/exception.jl

```diff
@@ -133,5 +133,8 @@ end
 function pyraise(e::PyError)
     ccall((@pysym :PyErr_Restore), Cvoid, (PyPtr, PyPtr, PyPtr),
           e.T, e.val, e.traceback)
-    e.T.o = e.val.o = e.traceback.o = C_NULL # refs were stolen
+    # refs were stolen
+    setfield!(e.T, :o, C_NULL)
+    setfield!(e.val, :o, C_NULL)
+    setfield!(e.traceback, :o, C_NULL)
```

Comment:
Upon reflection, I think it would be better to pass `pyincref(e.T), pyincref(e.val), pyincref(e.traceback)` to the `ccall` rather than setting the objects to null after the call

Reply:
Done.